### PR TITLE
Adds new Viro symptoms

### PR DESCRIPTION
**Pull request by CMOisLing**
###Intent of your Pull Request
I like the idea of double edged sword viro symptoms, they can either be extremely robust, or your doom depending on the application. Work in progress right now extremely unbalanced. Ill write more about it later promise, I can sleep easy knowing my master fork is clean now

Changelog
🆑
rscadd: New high level viro symptoms
wip: Putting this up so code bois can help me
/🆑

### CODE REVIEW

**Comment by ThatLing** [comment]
I like this but muh modular.

**Comment by ghost** [comment]
no more fucking heal symptoms i swear to god i will choke you

**Comment by CMOisLing** [comment]
I would say it’s more useful for killing people than healing currently

**Comment by CMOisLing** [comment]
What do you mean by modular, should I add some checks for non humans and animals

**Comment by ghost** [comment]
https://github.com/yogstation13/Yogstation-TG/blob/master/yogstation/README.md

Modularization.  Use the yogstation13 folder in there to add new procs etc.  If you need more help with it PM me on discord.

**Comment by CMOisLing** [comment]
For my "confusion" changes should I just put "Nerve Block" in a new file. Or do I have to do something else in that case. Im not gonna add anymore symptoms till I get my formating correct.

**Comment by CMOisLing** [comment]
I think I figured it out I’ll update it tomorrow

File: code/datums/diseases/advance/symptoms/confusion.dm [hunk 1]
```diff
@@ -33,7 +33,7 @@ Bonus
 					  <b>Transmission 6:</b> Increases confusion duration.<br>\
 					  <b>Stealth 4:</b> The symptom remains hidden until active."
 
-/datum/symptom/confusion/Start(datum/disease/advance/A)
+/datum/symptom/confusion/confusion/Start(datum/disease/advance/A)
```
Comment (Kmc2000) [inline code/datums/diseases/advance/symptoms/confusion.dm]: Wha

File: code/datums/diseases/advance/symptoms/confusion.dm [hunk 2]
```diff
@@ -43,7 +43,7 @@ Bonus
 	if(A.properties["stealth"] >= 4)
 		suppress_warning = TRUE
 
-/datum/symptom/confusion/Activate(datum/disease/advance/A)
+/datum/symptom/confusion/confusion/Activate(datum/disease/advance/A)
```
Comment (Kmc2000) [inline code/datums/diseases/advance/symptoms/confusion.dm]: Seriously what's with this

File: code/datums/diseases/advance/symptoms/heal.dm [hunk 3]
```diff
@@ -463,4 +463,4 @@
 	for(var/obj/item/bodypart/L in parts)
 		if(L.heal_damage(heal_amt/parts.len, heal_amt/parts.len))
 			M.update_damage_overlays()
-	return 1
+	return 1
```
Comment (Kmc2000) [inline code/datums/diseases/advance/symptoms/heal.dm]: return TRUE please

File: yogstation/code/datums/diseases/advance/symptoms/confusion.dm [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+			if(brain_damage)
+				M.adjustBrainLoss(3 * power, 80)
+				M.updatehealth()
+
+	return
+	
+/datum/symptom/confusion/numb
+
+	name = "Nerve Block"
+	desc = "The virus interferes with the proper function of the neural system, leading to bouts of confusion and erratic movement."
```
Comment (Kmc2000) [inline yogstation/code/datums/diseases/advance/symptoms/confusion.dm]: Wouldn't it be the nervous system that gets fucked up?

File: yogstation/code/datums/diseases/advance/symptoms/heal.dm [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+		M.adjustToxLoss(-5)
+	
+	if(prob(25))
+		to_chat(M, "<span class='notice'>You feel your wounds getting warm.</span>")
+
+	for(var/obj/item/bodypart/L in parts)
+		if(L.heal_damage(heal_amt/parts.len, heal_amt/parts.len))
+			M.update_damage_overlays()
+
+	return 1
```
Comment (Kmc2000) [inline yogstation/code/datums/diseases/advance/symptoms/heal.dm]: return TRUE (housekeeping)

**Review by Kmc2000 — COMMENTED** [review]
Good work, few things I can see though

**Comment by CMOisLing** [comment]
Thanks for prettying up my shitcode kmc. The changes to the tree where because nerve block and confusion where in the same file called confusion. I’m not sure if that was the right solution.

File: code/datums/diseases/advance/symptoms/heal.dm (hunk 3 above)
Comment (ThatLing) [reply code/datums/diseases/advance/symptoms/heal.dm]: Just delete the changes in this file

**Comment by ThatLing** [comment]
Delete the changes in the non-yogs files and this is good to go.

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 6]
```diff
@@ -657,4 +657,4 @@
 /obj/item/integrated_circuit/manipulation/matman/Destroy()
 	GET_COMPONENT(materials, /datum/component/material_container)
 	materials.retrieve_all()
-	.=..()
```
Comment (CMOisLing) [inline code/modules/integrated_electronics/subtypes/manipulation.dm]: WHAT AM I DOING WRONG I CANT REMOVE THIS SHIT

**Comment by CMOisLing** [comment]
It keeps readding the new line ive removed whats with that?

**Comment by CMOisLing** [comment]
The bot says im good? I guess its good to merge?

**Comment by ghost** [comment]
Huh it doesn't seem to pick up the removal of the newline.

**Comment by CMOisLing** [comment]
I’ve tried copying it straight from the master branch and it still says I added a line

**Comment by ghost** [comment]
Well best bet IMO is to delete the whole file, push that, then re-add it without the space straight from master.  Essentially reset what's not picking up the newline change.

**Comment by CMOisLing** [comment]
YES

**Comment by CMOisLing** [comment]
Alright ready to merge

**Comment by CMOisLing** [comment]
Need to change the levels to 6 and 5 I’ll do it when I get off work

**Comment by CMOisLing** [comment]
merge dat shit

File: yogstation/code/datums/diseases/advance/symptoms/confusion.dm [hunk 7]
```diff
@@ -0,0 +1,54 @@
+/datum/symptom/numb
+
+	name = "Nerve hardening"
+	desc = "The virus strengthens nerve connections decreasing interference to nerve connections, as a side effect the nervous system no longer reacts to pain."
```
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [inline yogstation/code/datums/diseases/advance/symptoms/confusion.dm]: Change this to be less stupid and it's good to go

**Review by 1fbff5f83b23d39d38b1dfcb4cac8d9b — CHANGES_REQUESTED** [review]

File: yogstation/code/datums/diseases/advance/symptoms/confusion.dm [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+		stamina_regen = TRUE
+		
+/datum/symptom/numb/Activate(datum/disease/advance/A)
+	if(!..())
+		return
+	var/mob/living/carbon/M = A.affected_mob
+	switch(A.stage)
+		if(1, 2, 3, 4)
+			if(prob(base_message_chance) && !suppress_warning)
+				to_chat(M, "<span class='notice'>[pick("You feel better.")]</span>")
```
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [inline yogstation/code/datums/diseases/advance/symptoms/confusion.dm]: Picking from a list with a single element

File: yogstation/code/datums/diseases/advance/symptoms/heal.dm [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+		
+	M.nutrition = max(M.nutrition - (Hunger_reduction * Hunger_multi), 0) // So heal to nutrient ratio doesnt change
+	
+	if(M.nutrition <= NUTRITION_LEVEL_STARVING && !Toxin_damage)
+		M.blood_volume -= 10
+		if(prob(45))
+			to_chat(M, "<span class='warning'>You feel like you are wasting away!</span>")
+	
+	if(Toxin_damage)
+		M.adjustToxLoss(-2)
```
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [inline yogstation/code/datums/diseases/advance/symptoms/heal.dm]: This HEALS toxin damage, is this what you wanted?

File: yogstation/code/datums/diseases/advance/symptoms/heal.dm [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+		return
+		
+	M.nutrition = max(M.nutrition - (Hunger_reduction * Hunger_multi), 0) // So heal to nutrient ratio doesnt change
+	
+	if(M.nutrition <= NUTRITION_LEVEL_STARVING && !Toxin_damage)
+		M.blood_volume -= 10
+		if(prob(45))
+			to_chat(M, "<span class='warning'>You feel like you are wasting away!</span>")
+	
+	if(Toxin_damage)
```
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [inline yogstation/code/datums/diseases/advance/symptoms/heal.dm:53]: This is never set to true anywhere

**Review by 1fbff5f83b23d39d38b1dfcb4cac8d9b — CHANGES_REQUESTED** [review]
Actually scratch that there's more things

File: yogstation/code/datums/diseases/advance/symptoms/heal.dm [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+	threshold_desc = "<b>Resistance 9:</b> No blood loss.<br>\
+					  <b>Stage Speed 7:</b> Reduces nutrients used."
+
+/datum/symptom/heal/conversion/Start(datum/disease/advance/A)
+	if(!..())
+		return
+	if(A.properties["stage_rate"] >= 7)
+		Hunger_reduction = 4
+	if(A.properties["resistance"] >= 9)
+		Toxin_damage = True
```
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [inline yogstation/code/datums/diseases/advance/symptoms/heal.dm]: TRUE, this ain't Python

**Review by 1fbff5f83b23d39d38b1dfcb4cac8d9b — APPROVED** [review]

File: yogstation/code/datums/diseases/advance/symptoms/heal.dm [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+	
+	if(M.nutrition <= NUTRITION_LEVEL_STARVING && !Toxin_damage)
+		M.blood_volume -= 10
+		if(prob(45))
+			to_chat(M, "<span class='warning'>You feel like you are wasting away!</span>")
+	
+	if(Toxin_damage)
+		M.adjustToxLoss(2)
+		if(prob(45))
+			to_chat(M, "<span class='warning'>You dont feel so well.</span>")
```
Comment (ATH1909) [inline yogstation/code/datums/diseases/advance/symptoms/heal.dm:56]: *don't

**Pull request merged**